server/src/main/java/io/crate/analyze/validator/GroupBySymbolValidator.java

```diff
@@ -34,23 +36,26 @@
     private static final InnerValidator INNER_VALIDATOR = new InnerValidator();
 
     public static void validate(Symbol symbol) throws IllegalArgumentException, UnsupportedOperationException {
-        symbol.accept(INNER_VALIDATOR, "Cannot GROUP BY '%s': invalid data type '%s'");
+        symbol.accept(INNER_VALIDATOR, false);
     }
 
-    private static class InnerValidator extends SymbolVisitor<String, Void> {
+    private static class InnerValidator extends SymbolVisitor<Boolean, Void> {
 
         @Override
-        public Void visitFunction(Function function, String errorMsgTemplate) {
+        public Void visitFunction(Function function, Boolean insideSubscriptFunction) {
             switch (function.type()) {
                 case SCALAR:
                     for (Symbol argument : function.arguments()) {
-                        argument.accept(this, errorMsgTemplate);
+                        argument.accept(this, function.name().equals(SubscriptFunction.NAME));
```

Comment:
I don't really understand why we have the special case for subscript functions. Shouldn't we be able to support all kinds of functions if subscripts work - or why are they special?

Reply:
actually you're right, pushed a fixup.